docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md

```diff
@@ -1,81 +1,133 @@
 # Handling Network Upgrades
 
-:::caution
-
-Following this guide when there is no network upgrade will result in you losing your place on the current network.
+Changes between the major consensus network versions are backward and forward
+breaking. You have to always run **a specific version of the Oasis Core to
+fetch and validate the blocks matching the specific consensus network version**.
+
+There are two kinds of consensus network upgrades that can occur:
+
+- *Seamless upgrade*: on-chain upgrade without resetting the consensus state or
+  changing the genesis document (e.g. [Testnet upgrade 2022-04-04], [Mainnet
+  upgrade 2021-08-31]).
+- *Dump & restore upgrade*: requires wiping state and starting the upgraded
+  network from a fresh genesis document (e.g. [Mainnet upgrade 2022-04-11
+  (Damask)], [Testnet upgrade 2022-03-03]).
+
+The specific Oasis Core version requirements also impact the way how you
+**initially sync your node with the network**:
+- If the last network upgrade was a dump & restore one, then your node will
+  complete the synchronization automatically by fetching and validating all
+  blocks following the state in the genesis document.
+- If the last network upgrade was a seamless one, you will first need to
+  download the older version of the Oasis Core to sync the initial blocks
+  and then sequentially perform seamless upgrade(s).
+
+For example, at time of writing this guide in order to sync your node from
+scratch on the testnet network you needed to do the following:
+
+- Download the genesis document and run Oasis Core 22.0.x which synced
+  blocks from epoch 14209 through (excluding) upgrade epoch 15056.
```

Comment:
Mention explicitly that one needs to wait until the node stops by itself for upgrade as noted below.